### PR TITLE
Dependency bumps to fix OWASP vulnerabilities

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR bumps the dependency versions of the gradle build to the latest versions to fix the latest OWASP vulnerabilities that are being logged in the nightly scan.

It's actually quite a big PR - bigger than what I anticipated, but I think thats a reflection of the fact we've not done this for a while (and perhaps should do more often)

Anyway, as well as the version bumps in the gradle build, theres a lot of "code changes" caused by linting errors (you're gonna love this @mattwills-moj-digital ! )
It seems one of the more recent changes in HMPPS Spring Boot gradle plugin has changed the linting rules! 😢 (either by bumping the version of the linter, or making a conscious decision to change them. Either way, I've had to make quite a few changes to the code to fix them!
The main (annoying) change is the positioning of comments!

Anyway, if you can get past the changes to comments and commas, then its just a PR that bumps a few versions! 👍

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -19,19 +19,21 @@ plugins {
...
-ext["logback.version"] = "1.4.14"
-ext["jackson-bom.version"] = "2.16.0"
+ext["jackson-bom.version"] = "2.16.1"
 
 allOpen {
   annotations(
     "javax.persistence.Entity",
     "javax.persistence.MappedSuperclass",
-    "javax.persistence.Embeddable"
+    "javax.persistence.Embeddable",
```
Comment (nathanrussell-moj-digital) [inline build.gradle.kts:36]: Linting rules now seems to apply to the gradle build, hence the need for a comma after the last item in a collection!

File: build.gradle.kts [hunk 2]
```diff
@@ -90,12 +92,12 @@ dependencies {
...
-  testFixturesImplementation("io.jsonwebtoken:jjwt-impl:0.11.5")
-  testFixturesImplementation("io.jsonwebtoken:jjwt-jackson:0.11.5")
-  testFixturesImplementation("com.nimbusds:nimbus-jose-jwt:9.31")
+  testFixturesImplementation("io.jsonwebtoken:jjwt-impl:$jsonWebTokenVersion")
+  testFixturesImplementation("io.jsonwebtoken:jjwt-jackson:$jsonWebTokenVersion")
+  testFixturesImplementation("com.nimbusds:nimbus-jose-jwt:$nimbusJwtVersion")
 
-  integrationTestImplementation("com.github.tomakehurst:wiremock-jre8-standalone:2.35.1")
-  testFixturesImplementation("com.github.tomakehurst:wiremock-jre8-standalone:2.35.1")
+  integrationTestImplementation("org.wiremock:wiremock-standalone:$wiremockVersion")
```
Comment (nathanrussell-moj-digital) [inline build.gradle.kts:99]: Wiremock's co-ordinates have changed with the v3 release.

File: domain/goal/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/GoalTest.kt [hunk 3]
```diff
@@ -22,7 +22,8 @@ class GoalTest {
     val goal = aValidGoal(steps = mutableListOf(step2, step3, step1)) // Steps passed into the goal out of sequence
 
     // Then
-    assertThat(goal.steps.map { it.title }).containsExactly( // steps are returned in correct sequence based on sequenceNumber
+    assertThat(goal.steps.map { it.title }).containsExactly(
```
Comment (nathanrussell-moj-digital) [inline domain/goal/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/GoalTest.kt:25]: This is the biggest (and most annoying IMHO) change in this PR.
Comments in a value list must now be on their own line, so everywhere we originally had a comment after the statement (if it's in a list of things), it now needs to be on it's own line! 😢 

For reference, this is the linting error message:
```
A comment in a 'value_argument_list' is only allowed when placed on a separate line (cannot be auto-corrected)
```
Theres some chat about it here: https://github.com/pinterest/ktlint/issues/2445

File: domain/induction/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/service/InductionServiceTest.kt [hunk 4]
```diff
@@ -35,7 +35,7 @@ class InductionServiceTest {
   private lateinit var inductionEventService: InductionEventService
 
   companion object {
-    private const val prisonNumber = "A1234AB"
+    private const val PRISON_NUMBER = "A1234AB"
```
Comment (nathanrussell-moj-digital) [inline domain/induction/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/service/InductionServiceTest.kt:38]: consts now have to be upper case snake case. I'm OK with this one TBH 🤷

**Review by mattwills-moj-digital — APPROVED** [review]
Painful! LGTM 👍

**Pull request merged**